skimage/transform/tests/test_histogram_matching.py

```diff
@@ -0,0 +1,104 @@
+import numpy as np
+
+from skimage.transform import histogram_matching
+from skimage import transform
+from skimage import data
+
+from skimage._shared.testing import assert_array_equal, \
+    assert_array_almost_equal, assert_almost_equal
+
+import pytest
+
+
+def test_get_separate_channels_single():
+    # given
+    image = np.ones((4, 5))
+    expected_channels = np.ones((1, 4, 5))
+
+    # when
+    channels = histogram_matching._get_separate_channels(image)
+
+    # then
+    assert_array_equal(channels, expected_channels)
+
+
+def test_get_separate_channels_multiple():
+    # given
+    image = np.full((4, 5, 3), [1, 0, 3])  # channels of zeros, ones and threes
+    expected_channels = np.asarray([
+        np.ones((4, 5)),
+        np.zeros((4, 5)),
+        np.full((4, 5), 3)
+    ])
+
+    # when
+    channels = histogram_matching._get_separate_channels(image)
+
+    # then
+    assert_array_equal(channels, expected_channels)
+
+
+@pytest.mark.parametrize('array, template, expected_array', [
+    (np.arange(10), np.arange(100), np.arange(9, 100, 10)),
+    (np.random.rand(4), np.ones(3), np.ones(4))
+])
+def test_match_array_values(array, template, expected_array):
+    # when
+    matched = histogram_matching._match_array_values(array, template)
+
+    # then
+    assert_array_almost_equal(matched, expected_array)
+
+
+def _calculate_image_empirical_pdf(image):
+    """Helper function for calculating empirical probability density function
+    of a given image for all channels"""
+
+    channels = histogram_matching._get_separate_channels(image)
+    channels_pdf = []
+    for channel in channels:
+        channel_values, counts = np.unique(channel, return_counts=True)
+        channel_quantiles = np.cumsum(counts).astype(np.float64)
+        channel_quantiles /= channel_quantiles[-1]
+
+        channels_pdf.append((channel_values, channel_quantiles))
+
+    return np.asarray(channels_pdf)
+
+
+image_rgb = data.chelsea()
+template_rgb = data.astronaut()
+
+
+@pytest.mark.parametrize('image, reference', [
+    (image_rgb[:, :, 0], template_rgb[:, :, 0]),
+    (image_rgb, template_rgb)
+])
+def test_match_histograms(image, reference):
```

Comment:
A suggestion: You might want to combine unit tests for the same function in a class. In this case you could save the global variables `image_rgb` and `template_rgb` as class attributes which may be a more contained solution.

Reply:
Good point, especially with the helper function. 
Moved functional tests to separate function `TestMetchHistogram`.